rasa_core/server.py

```diff
@@ -822,8 +822,53 @@ def validate_request_body(request: Request, error_message: Text):
         logger.debug("Successfully loaded model '{}'.".format(model_path))
         return response.json(None, status=204)
 
+    @app.get("/conversations/<sender_id>/messages")
+    @requires_auth(app, auth_token)
+    async def retrieve_conversation_messages(request: Request, sender_id: Text):
+        """Get only actual conversation messages ('user' and 'bot' events) from tracker."""
+
+        if not app.agent.tracker_store:
+            raise ErrorResponse(503, "NoTrackerStore",
+                                "No tracker store available. Make sure to "
+                                "configure a tracker store when starting "
+                                "the server.")
+
+        # parameters
+        default_verbosity = EventVerbosity.ALL
+
+        verbosity = event_verbosity_parameter(request,
+                                              default_verbosity)
+
+        # retrieve tracker and set to requested state
+        tracker = app.agent.tracker_store.get_or_create_tracker(sender_id)
+        if not tracker:
+            raise ErrorResponse(503,
+                                "NoDomain",
+                                "Could not retrieve tracker. Most likely "
+                                "because there is no domain set on the agent.")
+
+        # get current state of a tracker and then extract events from it
+        state = tracker.current_state(verbosity)
+        all_events = state["events"]
+
+        # retrieve only 'user' and 'bot' events from the complete list of events
+        user_bot_kv_list = ["user", "bot"]
+        all_user_bot_events = [e for e in all_events if e["event"] in user_bot_kv_list]
+
+        # retrieve only 'events', 'timestamp' and 'text' part of the 'user' and 'bot' events
+        user_bot_keys_list = ["event", "timestamp", "text"]
+        user_bot_events = [dict((key, value) for key, value in a.items() if key in user_bot_keys_list)
```

Comment:
👍  as a thought one could add the switch betweem event and author here.
```
def set_key_value(key):
    if key == 'event':
        return 'author'
    else:
          return key. 
```
Then add this in.   dict (set_key_value(key), value)

Reply:
ok. I can do this change :)